File: kraken/purpur/examples/java/PlaceholderBlockFeature.java
package org.hexalite.network.kraken.examples.java;

import org.hexalite.network.kraken.extension.ToolLevel;
import org.hexalite.network.kraken.extension.ToolType;
import org.hexalite.network.kraken.gameplay.feature.block.CustomBlockFeature;
import org.hexalite.network.kraken.gameplay.feature.item.CustomItemFeature;

public class PlaceholderBlockFeature extends CustomBlockFeature {
    public static int ID = 1;

    static PlaceholderBlockFeature BLOCK = new PlaceholderBlockFeature();
    static PlaceholderBlockFeature.Item ITEM = new PlaceholderBlockFeature.Item();

    PlaceholderBlockFeature() {
        super(ID);
        withHardness(20, ToolLevel.Iron, ToolType.Pickaxe);
    }

    public static class Item extends CustomItemFeature {
        Item() {
            super(ID);
            withLocalizedName("block.hexalite.placeholder");
            withAttack(1.0, 7.0);
        }
    }

    public static PlaceholderBlockFeature block() {
        return BLOCK;
    }

    public static PlaceholderBlockFeature.Item item() {
        return ITEM;
    }
}

File: arcade/duels/src/main/kotlin/org/hexalite/network/duels/gameplay/feature/block/PlaceholderBlockFeature.kt
package org.hexalite.network.duels.gameplay.feature.block

import org.hexalite.network.kraken.extension.ToolLevel
import org.hexalite.network.kraken.extension.ToolType
import org.hexalite.network.kraken.gameplay.feature.block.CustomBlockFeature
import org.hexalite.network.kraken.gameplay.feature.block.setToolBasedHardness
import org.hexalite.network.kraken.gameplay.feature.item.CustomItemFeature

object PlaceholderBlockFeature: CustomBlockFeature(1) {
    init {
        setToolBasedHardness(20, ToolLevel.Iron, ToolType.Pickaxe)
    }
}

object PlaceholderBlockItemFeature: CustomItemFeature(1) {
    init {
        name(key = "block.hexalite.placeholder")
        attack(speed = 1.0, damage = 7.0)
    }
}
